src/components/store-deploy-block.tsx

```diff
@@ -46,24 +50,48 @@ export const StoreDeployBlock: FunctionComponent<DocumentStoreAddressProps> = ({
         setLog(
           `Document Store Deployed. Find more details at <a href="${etherscanNetwork}/address/${transaction.contractAddress}" target="_blank">${etherscanNetwork}/address/${transaction.contractAddress}</a>.`
         );
+        setDocumentStoreInformation(documentStoreAddress);
         validateStorageAddress(transaction.contractAddress);
         setShowModal(false);
       }
     }
     setProcessing(false);
   };
 
+  useEffect(() => {
+    async function getLocalStorageDocumentStore() {
+      const documentStoreInformation = await retrieveDocumentStoreInformation();
+      setLocalDocumentStores(documentStoreInformation);
+    }
+
+    getLocalStorageDocumentStore();
+  }, []);
+
+  const onSuggestionsFetchRequested = async (value: string, reason: string): Promise<void> => {
+    setDocumentStoreAddress(value);
+    if (reason == "input-focused") {
```

Comment:
why do we do this block considering we already get the document store above, in the useEffect that execute once ?

Reply:
This is used to fetch new document store information when the user changes Metamask account. I will remove the useEffect.